### PR TITLE
Add a configuration option to allow shared bookmarks: `$cfg['AllowSharedBookmarks']`

### PR DESCRIPTION
**Pull request by erkia**
### Description

In some setups, it is desirable to prevent users from creating shared SQL bookmarks. 

This pull request adds a configuration option that can be used to allow/disallow shared bookmarks feature.

### CODE REVIEW

File: doc/config.rst [hunk 1]
```diff
@@ -3527,6 +3527,13 @@ Various display setting
     specify the amount of saved history items using
     :config:option:`$cfg['QueryHistoryMax']`.
 
+.. config:option:: $cfg['AllowSharedBookmarks']
+
+    :type: boolean
+    :default: true
+
```
Comment (williamdes) [inline doc/config.rst:3534]: Please add a `versionadded: 6.0.0`
There is examples in this file

File: config.sample.inc.php [hunk 2]
```diff
@@ -137,6 +137,13 @@
  */
 //$cfg['QueryHistoryMax'] = 100;
 
+/**
```
Comment (williamdes) [inline config.sample.inc.php:140]: I am not sure it's worth adding it to the example file

Comment (erkia) [reply config.sample.inc.php:140]: Indeed, probably not - will remove.

**Review by williamdes — APPROVED** [review]
This looks good, thank you !

**Review by MauricioFauth — APPROVED** [review]

**Comment by erkia** [comment]
Rebased, adapted variable name changes in `libraries/classes/Bookmark.php`. Failing tests seem not to be related to these changes.

**Review by williamdes — APPROVED** [review]
@MauricioFauth could you have a look to this one for a merge ?

**Comment by williamdes** [comment]
Thank you @erkia for this nice /security/ feature 🎉

**Pull request merged**